instrumentation/net/http/otelhttp/config.go

```diff
@@ -0,0 +1,180 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelhttp
+
+import (
+	"net/http"
+
+	"go.opentelemetry.io/contrib"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	instrumentationName = "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+// config represents the configuration options available for the http.Handler
+// and http.Transport types.
+type config struct {
+	Tracer            trace.Tracer
+	Meter             metric.Meter
+	Propagators       propagation.TextMapPropagator
+	SpanStartOptions  []trace.SpanOption
+	ReadEvent         bool
+	WriteEvent        bool
+	Filters           []Filter
+	SpanNameFormatter func(string, *http.Request) string
+
+	TracerProvider trace.TracerProvider
+	MeterProvider  metric.MeterProvider
+}
+
+// Option Interface used for setting *optional* config properties
+type Option interface {
+	Apply(*config)
+}
+
+// OptionFunc provides a convenience wrapper for simple Options
+// that can be represented as functions.
+type OptionFunc func(*config)
+
+func (o OptionFunc) Apply(c *config) {
+	o(c)
+}
+
+// newConfig creates a new config struct and applies opts to it.
+func newConfig(opts ...Option) *config {
+	c := &config{
+		Propagators:    otel.GetTextMapPropagator(),
+		TracerProvider: otel.GetTracerProvider(),
+		MeterProvider:  global.GetMeterProvider(),
+	}
+	for _, opt := range opts {
+		opt.Apply(c)
+	}
+
+	c.Tracer = c.TracerProvider.Tracer(
+		instrumentationName,
+		trace.WithInstrumentationVersion(contrib.SemVersion()),
+	)
+	c.Meter = c.MeterProvider.Meter(
+		instrumentationName,
+		metric.WithInstrumentationVersion(contrib.SemVersion()),
+	)
+
+	return c
+}
+
+// WithNop returns a no operation option.
+// It can be used for extensibilty.
+func WithNop() Option {
+	return OptionFunc(func(*config) {
+	})
+}
```

Comment:
This is the only change in OTel SDK. This is a way that requires the fewest code changes in OTel SDK.

I think a better alternative may be to export the `config` type and at the same time unexport all fields in the `config` struct. It would give the possibility to implement the `Option` interface which right now was not really possible (without the `WithNop` hack).

Any other ideas?